imap_data_access/processing_input.py

```diff
@@ -168,6 +211,11 @@ class ScienceInput(ProcessingInput):
      and descriptor.
     """
 
+    class InvalidScienceInputError(Exception):
```

Comment:
Personally, I think this is too many errors, can we just have a single `ProcessingInputError` that we catch everywhere rather than needing to catch all of these separately? The message to each when we raise can be different if we want.